selfservice/strategy/oidc/strategy_login.go

```diff
@@ -109,7 +105,12 @@ func (s *Strategy) processLogin(w http.ResponseWriter, r *http.Request, a *login
 			}
 
 			// This flow only works for browsers anyways.
-			aa, err := s.d.RegistrationHandler().NewRegistrationFlow(w, r, flow.TypeBrowser, opts...)
+			aa, err := s.d.RegistrationHandler().NewRegistrationFlow(w, r, a.Type, opts...)
```

Comment:
I would have expected a few tests in Go (strategy_test) that cover this behavior. They're also going to execute faster than the playwright tests. It is a bit of duplication though, but it makes debugging much simpler. WDYT? Worth the effort?

Reply:
Definitely, it would have made debugging much simpler during development as well! I just didn't find the right spot before, but now I added the tests. PTAL :)